.style-and-markdown-build.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
```

Comment:
Don't just continue if errors happen e.g. when changing directory:
```suggestion

set -eu

```

Reply:
Not sure if you missed this. But without this, the script will run successfully with a return value of `0` even if the `mkdocs` command fails.

Reply:
It shouldn't, that's what the `ret` flag is for.  I'd like it to run all the tests, and then fail if any of them have failed.

But now that you mention it, there's a typo in the commented out code :D

Reply:
> It shouldn't, that's what the `ret` flag is for. I'd like it to run all the tests, and then fail if any of them have failed.

Yes it will and the `ret` will not help you a bit.
Let me give you one example (there are lots of other things which can go wrong. Don't try to catch them yourself):

```
% bash .style-and-markdown-build.sh
Markdown doc admin build log:
.style-and-markdown-build.sh: line 43: mkdocs: command not found
Markdown doc developer build log:
.style-and-markdown-build.sh: line 43: mkdocs: command not found
Markdown doc user build log:
.style-and-markdown-build.sh: line 43: mkdocs: command not found

> docs@ test /home/lars/dev/opencast/docs/guides
> markdownlint *.md ./admin/**/*.md ./developer/**/*.md ./user/**/*.md ../../*.md

% echo $?
0
```

As you can see, `mkdocs` clearly failed since the command did not exist but your script returned success.
Your script currently works fundamentally different than the tests did before since you are just looking at the specific tests but knowingly ignore all errors happening with the commands itself. The `set -e` will change that. It has no effect on the `ret` values and their evaluation you built.

Reply:
Good point, added.